test-app/src/api-test.ts

```diff
@@ -0,0 +1,172 @@
+import {
+  App,
+  aws_apigateway,
+  aws_dynamodb,
+  aws_logs,
+  Stack,
+} from "aws-cdk-lib";
+import {
+  ApiIntegrations,
+  ExpressStepFunction,
+  Function,
+  SyncExecutionSuccessResult,
+  Table,
+} from "functionless";
+
+export const app = new App();
+
+const stack = new Stack(app, "api-test-app-stack");
+
+const restApi = new aws_apigateway.RestApi(stack, "api", {
+  restApiName: "api-test-app-api",
+});
+
+interface FnRequest {
+  pathParameters: {
+    num: number;
+  };
+  queryStringParameters: {
+    str: string;
+  };
+  body: {
+    bool: boolean;
+  };
+}
+
+const fn = new Function(
+  stack,
+  "fn",
+  async (event: { inNum: number; inStr: string; inBool: boolean }) => ({
+    fnNum: event.inNum,
+    fnStr: event.inStr,
+    fnBool: event.inBool,
+    nested: {
+      again: {
+        num: 123,
+      },
+    },
+  })
+);
+
+const fnResource = restApi.root.addResource("fn").addResource("{num}");
+const fnIntegration = ApiIntegrations.aws({
+  request: (req: FnRequest) =>
+    fn({
+      inNum: req.pathParameters.num,
+      inStr: req.queryStringParameters.str,
+      inBool: req.body.bool,
+    }),
+  response: (resp) => ({
+    resultNum: resp.fnNum,
+    resultStr: resp.fnStr,
+    nested: resp.nested.again.num,
+  }),
+  errors: {
+    400: () => ({ msg: "400" }),
+  },
+});
+fnIntegration.addMethod("POST", fnResource);
+
+const sfn = new ExpressStepFunction(
+  stack,
+  "express-sfn",
+  {
+    logs: {
+      destination: new aws_logs.LogGroup(stack, "express-sfn-logs"),
+      includeExecutionData: true,
+    },
+  },
+  (input: { num: number; str: string }) => ({
+    sfnNum: input.num,
+    sfnStr: input.str,
+  })
+);
+
+interface MockRequest {
+  pathParameters: {
+    num: 200 | 500;
+  };
+}
+
+const mockResource = restApi.root.addResource("mock").addResource("{num}");
+const mock = ApiIntegrations.mock({
+  request: (req: MockRequest) => ({
+    statusCode: req.pathParameters.num,
+  }),
+  responses: {
+    200: () => ({
+      body: {
+        num: 12345,
+      },
+    }),
+    500: () => ({
+      msg: "error",
+    }),
+  },
+});
+mock.addMethod("GET", mockResource);
```

Comment:
This looks like the method is added to the mock. But is this the mock adding itself as a method on mockResource? Should it be reversed?

```ts
mockResource.addMethod("GET", mock):
```

Reply:
this is the same as `AppsyncResolver`. it's going to change